pytest_asyncio/plugin.py

```diff
@@ -663,7 +670,12 @@ def wrap_in_sync(
     @functools.wraps(func)
     def inner(*args, **kwargs):
         coro = func(*args, **kwargs)
-        _loop = _get_event_loop_no_warn()
+        try:
+            _loop = _get_event_loop_no_warn()
+        except RuntimeError:
+            # Handle situation where asyncio.set_event_loop(None) removes shared loops.
+            _reinstate_event_loop_on_main_thread()
```

Comment:
The function `_get_event_loop_no_warn` is a helper that suppresses a call to `get_event_loop`, which is effectively deprecated in favor of `get_running_loop`. This deprecation was one of the drivers for removing the `event_loop` fixture, which relied on this functionality.

The fact that `wrap_in_sync` still uses this is unfortunate and I hope to rectify this in #1203.

The point I'm trying to make is that *wrap_in_sync* seems to be the wrong place to reinstate the loop, because it would mean doubling down on the *get_event_loop* use.